lib/whitehall_importer/import.rb

```diff
@@ -61,8 +61,22 @@ def create_document(user_ids)
         created_at: whitehall_document["created_at"],
         updated_at: whitehall_document["updated_at"],
         created_by_id: user_ids[event["whodunnit"]],
+        first_published_at: first_publish_date,
         imported_from: "whitehall",
       )
     end
+
+    def first_publish_date
+      first_publish_event = history(first_edition).first_state_event("published") || {}
+      first_publish_event["created_at"]
+    end
+
+    def first_edition
+      whitehall_document["editions"].first
+    end
+
+    def history(edition)
+      @history ||= EditionHistory.new(edition["revision_history"])
+    end
```

Comment:
You can't use a memo with an argument as it'll be ignored on subsequent calls.

Since you're only accessing first_edition for history you might just want to have a first_edition_history method and drop the first_edition one?

Reply:
Good spot!  I've updated to have a `first_edition_history` method instead